### PR TITLE
test: enable running tests under root user

### PR DESCRIPTION
**Pull request by liubin**
Add tests that run under root user to test special cases.

Signed-off-by: bin <bin@hyper.sh>

### CODE REVIEW

**Comment by egernst** [comment]
I see the static checks passing, which I assume is good? I didn't realize our CI wasn't covering unit tests which required root! Thanks for the PR!

**Comment by liubin** [comment]
hi  @egernst there are two tests(in file `kata-check_amd64_test.go`) are skipped now, I will fix them later.

File: src/runtime/cli/kata-check.go [hunk 1]
```diff
@@ -221,15 +221,20 @@ func checkKernelModules(modules map[string]kernelModule, handler kernelParamHand
 		return 0, err
 	}
 
+	fmt.Printf("AAAAAA onVMM %+v\n", onVMM)
+	fmt.Printf("BBBBB onVMM %+v\n", modules)
```
Comment (sameo) [inline src/runtime/cli/kata-check.go:225]: I assume this is debugging leftovers?

File: src/runtime/cli/kata-check.go [hunk 2]
```diff
@@ -221,15 +221,20 @@ func checkKernelModules(modules map[string]kernelModule, handler kernelParamHand
...
+
 	for module, details := range modules {
 		fields := logrus.Fields{
 			"type":        "module",
 			"name":        module,
 			"description": details.desc,
 		}
+		fmt.Printf("BBBBB module %+v\n", module)
+		fmt.Printf("BBBBB details %+v\n", details)
```
Comment (sameo) [inline src/runtime/cli/kata-check.go:234]: ditto?

File: src/runtime/cli/kata-check.go (hunk 1 above)
Comment (liubin) [reply src/runtime/cli/kata-check.go:225]: Yes, the test is not passed yet.

**Pull request closed without merging**